File: src/main/java/de/steklopod/sec04/Lec01Handle.java
package de.steklopod.sec04;

import de.steklopod.utils.Util;
import reactor.core.publisher.Flux;

public class Lec01Handle {

    public static void main(String[] args) {

        // handle = filter + map
        Flux.range(1, 20)
                .handle((integer, synchronousSink) -> {
                    if(integer == 7)
                        synchronousSink.complete();
                    else
                        synchronousSink.next(integer);
                })
                .subscribe(Util.subscriber());

    }

}

File: src/main/kotlin/de/steklopod/sec04/Lec01Handle.kt
package de.steklopod.sec04

import de.steklopod.utils.Util.subscriber
import reactor.core.publisher.Flux
import reactor.core.publisher.SynchronousSink

object Lec01Handle {
    @JvmStatic
    fun main(args: Array<String>) {

        // handle = filter + map
        Flux.range(1, 20)
            .handle { integer: Int, synchronousSink: SynchronousSink<Any> ->
                if (integer == 7) synchronousSink.complete()
                else synchronousSink.next(integer)
            }
            .subscribe(subscriber())
    }
}
